File: Evergrande/src/gestorAplicación/Cita.java
package gestorAplicación;

public class Cita {
	private static int idCitas = 0;
	private int idCita;
	private int dia;
	private int mes;
	private int ano;
	private String hora;
	
	public Cita(int ano, int mes, int dia, String hora) {
		this.ano = ano;
		this.mes = mes;
		this.dia = dia;
		this.hora = hora;
		this.idCita = Cita.idCitas;
		Cita.idCitas++;
	}
	
	public int getIdCita() {
		return idCita;
	}
	
	public void setIdCita(int idCita) {
		this.idCita = idCita;
	}
	
	public int getDia() {
		return dia;
	}
	
	public void setDia(int dia) {
		this.dia = dia;
	}
	
	public int getMes() {
		return mes;
	}
	
	public void setMes(int mes) {
		this.mes = mes;
	}
	
	public int getAno() {
		return ano;
	}
	
	public void setAno(int ano) {
		this.ano = ano;
	}
	
	public String getHora() {
		return hora;
	}
	
	public void setHora(String hora) {
		this.hora = hora;
	}
}

File: Python/src/gestorAplicacion/otros/Cita.py
class Cita:
    _citas = []

    def __init__(self, dia, mes, ano, hora, idAgente, idInmueble):
        self._dia = dia
        self._mes = mes
        self._ano = ano
        self._hora = hora
        self._idAgente = idAgente
        self._idInmueble = idInmueble

        try:
            self._idCita = None
            self._idCita = Cita._citas[-1]._idCita + 1
        except:
            self._idCita = 0

        Cita._citas.append(self)

    @classmethod
    def cancelar(cls, idCita):
        for i in range(len(cls._citas)):
            if cls._citas[i]._idCita == idCita:
                cls._citas.pop(i)
                break
    
    @classmethod
    def getCitaByID(cls, id):
        return list(filter(lambda cita: cita._idCita == id, cls._citas))[0]

    @classmethod
    def existeCita(cls, id):
        return len(list(filter(lambda cita: cita._idCita == id, cls._citas))) > 0

    def __str__(self):
        return ""

    # getters y setters
    @classmethod
    def getCitas(cls):
        return cls._citas

    @classmethod
    def setCitas(cls, citas):
        cls._citas = citas

    def getIdCita(self):
        return self._idCita

    def setIdCita(self, idCita):
        self._idCita = idCita

    def getDia(self):
        return self._dia

    def getMes(self):
        return self._mes

    def getAno(self):
        return self._ano

    def getHora(self):
        return self._hora

    def getIdAgente(self):
        return self._idAgente

    def getIdInmueble(self):
        return self._idInmueble

    def setDia(self, dia):
        self._dia = dia

    def setMes(self, mes):
        self._mes = mes

    def setAno(self, ano):
        self._ano = ano

    def setHora(self, hora):
        self._hora = hora

    def setIdAgente(self, idAgente):
        self._idAgente = idAgente

    def setIdInmueble(self, idInmueble):
        self._idInmueble = idInmueble
